delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,497 @@
+
+    def __init__(self, company):
+        self.default_lang = company.partner_id.lang or "en"
+
+    def _get_language(self, lang):
+        """ Return a language to iso format from odoo format.
+
+        `iso_code` field in res.lang is not mandatory thus not always set.
+        Use partner language if available, otherwise use english
+
+        :param partner: partner browse record
+        :return: language code to use.
+
+        """
+        if not lang:
+            lang = self.default_lang
+        available_languages = ["de", "en", "fr", "it"]  # Given by API doc
+        lang_code = lang.split("_")[0]
+        if lang_code in available_languages:
+            return lang_code
+        return "en"
+
+    def _prepare_recipient(self, picking):
+        """ Create a ns0:Recipient as a dict from a partner
+
+        :param partner: partner browse record
+        :return a dict containing data for ns0:Recipient
+
+        """
+        partner = picking.partner_id
+
+        partner_name = partner.name or partner.parent_id.name
+        recipient = {
+            "name1": partner_name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "email": partner.email or None,
+        }
+
+        if partner.country_id.code:
+            recipient["country"] = partner.country_id.code.upper()
+
+        if partner.street2:
+            recipient["addressSuffix"] = partner.street2
+
+        company_partner_name = partner.commercial_company_name
+        if company_partner_name and company_partner_name != partner_name:
+            recipient["name2"] = partner.parent_id.name
+            recipient["personallyAddressed"] = False
+
+        # Phone and / or mobile should only be displayed if instruction to
+        # Notify delivery by telephone is set
+        postlogistics_partner_codes = partner.postlogistics_option_ids.mapped("code")
+
+        is_phone_required = "ZAW3213" in postlogistics_partner_codes
+
+        if is_phone_required:
+            phone = picking.delivery_phone or partner.phone
+            if phone:
+                recipient["phone"] = phone
+
+            mobile = picking.delivery_mobile or partner.mobile
+            if mobile:
+                recipient["mobile"] = mobile
+
+        return recipient
+
+    def _prepare_customer(self, picking):
+        """ Create a ns0:Customer as a dict from picking
+
+        This is the Postlogistic Customer, thus the sender
+
+        :param picking: picking browse record
+        :return a dict containing data for ns0:Customer
+
+        """
+        company = picking.company_id
+        partner = company.partner_id
+
+        customer = {
+            "name1": partner.name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "domicilePostOffice": picking.carrier_id.postlogistics_office,
```

Comment:
When not used, must be `None` instead of `False`. This seems the rule for the strings other "False" is printed on the label
```suggestion
            "domicilePostOffice": picking.carrier_id.postlogistics_office or None,
```

Reply:
updated in https://github.com/OCA/delivery-carrier/pull/323/commits/3b74e94c38f331ff5ff54cb2e1c0525991379f44